src/utils.rs

```diff
@@ -538,33 +539,29 @@ pub async fn suspend_unpaid_happs(
 
     for invoice in &pending_transactions.invoice_pending {
         if let Some(POS::Hosting(_)) = &invoice.proof_of_service {
-            if let Some(expiration_date) = invoice.expiration_date {
-                if expiration_date.as_millis() < Utc::now().timestamp_millis() {
-                    if let Some(note) = invoice.note.clone() {
-                        let invoice_note: Result<InvoiceNote, _> = serde_yaml::from_str(&note);
-                        match invoice_note {
-                            Ok(note) => {
-                                let hha_id = note.hha_id;
-                                suspended_happs.push(hha_id.clone().to_string());
-                                core_app_client
-                                    .app
-                                    .zome_call_typed(
-                                        CoreAppRoleName::HHA.into(),
-                                        ZomeName::from("hha"),
-                                        FunctionName::from("disable_happ"),
-                                        ExternIO::encode(HappAndHost {
-                                            happ_id: hha_id.clone(),
-                                            holoport_id: holoport_id.to_string(),
-                                        })?,
-                                    )
-                                    .await?;
-                            }
-                            Err(e) => {
-                                error!("Error parsing invoice note: {:?}", e);
-                            }
-                        }
+            let note = invoice.clone().get_note();
+            match note {
+                Some(note) => {
+                    if note.invoice_due_date <= Timestamp(Utc::now().timestamp()) {
+                        let hha_id = note.hha_id;
+                        suspended_happs.push(hha_id.clone().to_string());
+                        core_app_client
+                            .app
+                            .zome_call_typed(
+                                CoreAppRoleName::HHA.into(),
+                                ZomeName::from("hha"),
+                                FunctionName::from("disable_happ"),
+                                ExternIO::encode(HappAndHost {
+                                    happ_id: hha_id.clone(),
+                                    holoport_id: holoport_id.to_string(),
+                                })?,
+                            )
+                            .await?;
                     }
                 }
+                None => {
+                    error!("cannot get note from transaction")
```

Comment:
If this error occurs, it means that hosting service was provided and invoiced, but that an internal error occurred in our system during the creation or reading of the note.  This means that an outstanding balance exists.  

How are a planning on surfacing the service ops and/or customer so we can mitigate?  Should we plan on adding a sentry notification or dedicated mm channel?

Reply:
Let's have a chat about this :)